### PR TITLE
Bump hwi to 0.4.0

### PR DESCRIPTION
**Pull request by danielabrozzoni**
### Changelog notice

- Bump hwi to 0.4.0

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

File: examples/hardware_signer.rs [hunk 1]
```diff
@@ -23,26 +24,27 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
...
-    let descriptors = client.get_descriptors(None)?;
+    let descriptors = client.get_descriptors::<Descriptor<DescriptorPublicKey>>(None)?;
     println!(
         "The hardware wallet's descriptor is: {}",
         descriptors.receive[0]
     );
 
     // Creating a custom signer from the device
-    let custom_signer = HWISigner::from_device(first_device, HWIChain::Test)?;
+    let custom_signer = HWISigner::from_device(&first_device, HWIChain::Test)?;
```
Comment (rajarshimaitra) [inline examples/hardware_signer.rs:44]: I am hitting an `OSError`  at this line with my coldcard.. Any guess?

```
Hold tight, I'm connecting to your hardware wallet...
No module named 'hwilib.devices.jadepy.jade_ble'
BLE scanning/connectivity will not be available
Look what I found, a coldcard!
The hardware wallet's descriptor is: pkh([ef2dfe99/44'/1'/0']tpubDDEc4pGkD8MPWq2D1DGb3DWH2RcXTe9cZYGqgFBGwYRA7rgf1uZop65ptiiKCKB12vS4UrysKucmdjFwjhVzpwNmcVNBXd84uqtj8TYLuVx/0/*)#ktdw6k4n
Error: PyErr("OSError: open failed")
```

**Review by rajarshimaitra — COMMENTED** [review]
Review ACK.. 

I am thinking how about adding a link to https://github.com/bitcoin-core/HWI in the doc above for people who don't know where to get the hwi python package from..

**Comment by notmandatory** [comment]
Should we try to get this into the 0.26.0 release? The code freeze was supposed to be today..but will probably delayed a few days..

**Comment by danielabrozzoni** [comment]
>  Should we try to get this into the 0.26.0 release? The code freeze was supposed to be today..but will probably delayed a few days..

I think it'd be cool if we can make it :)

@rajarshimaitra can you try with bdk from master? It's possible that you'll hit the same error, in which case it's definitely to fix, but at least we know that it wasn't introduced in this PR :)

**Comment by rajarshimaitra** [comment]
> @rajarshimaitra can you try with bdk from master? It's possible that you'll hit the same error, in which case it's definitely to fix, but at least we know that it wasn't introduced in this PR :)

Yes I am hitting the same one in master too..

**Comment by notmandatory** [comment]
@rajarshimaitra since you're getting the same error on `master` branch this should be ready to go right?

**Review by notmandatory — APPROVED** [review]
ACK 4cad18bbcaa8199b82c10291aa38941f06823568

**Comment by danielabrozzoni** [comment]
> @rajarshimaitra since you're getting the same error on master branch this should be ready to go right?


I think it is! :)

**Pull request merged**